service/rds/api_client.go

```diff
@@ -194,3 +195,56 @@ func addRequestIDRetrieverMiddleware(stack *middleware.Stack) {
 func addResponseErrorMiddleware(stack *middleware.Stack) {
 	awshttp.AddResponseErrorMiddleware(stack)
 }
+
+/*
+ Handwritten code
+*/
+
+// PresignClient is a presigner client
+type PresignClient struct {
+	// no need to export these members
+	// client used by presigner
+	client *Client
+	// signer used for presigning
+	presigner *v4.Signer
+}
+
+// NewPresignClient returns a pre-signed client
+func NewPresignClient(options Options, optFns ...func(*Options)) *PresignClient {
+	client := New(options, optFns...)
+	presigner := v4.NewSigner()
+	return &PresignClient{
+		client:    client,
+		presigner: presigner,
+	}
+}
```

Comment:
If using the PresignOptions type with the constructor this would be adjust to take that value instead of API client's Options.

##### Taking PresignOption with optional client parameter.

```go
client := rds.New(options)

// e.g. PresignClient creating it own API client with the options provided
presigner := rds.NewPresignClient(options,  func(o *rds.PresignOptions) {
     o.Presigner = v4.NewPresigner(/*signer options*/)
})

// Create presign client from API client with client options for region and presigner options
presigner := rds.NewPresignClientWithClient(client, func(o *rds.PresignOptions) {
     o.Presigner = v4.NewPresigner()
     o.OptionFns = append(o.OptionFns, func(o*Options){ o.Region = "us-west-2" })
})

// To use the presign client from a client with options, need to merge presign options together, before creating 
// copy of the client.
func NewPresignClientWithClient(client *Client, optFns ...func(*PresignOptions)) *PresignClient {
       var presignOptions PresignOptions
       for _, fn := range optFns {
            fn(&presignOptions)
       }

       client = client.copyAPIClient(presignOptions.OptionsFns....)
}

// If user calls NewPresignClientWithClient and provides functional options need to create
// a copy of the API client merging in the new options.
func (c *Client) copyAPIClient(optFns ...func(*Option)) *Client {
      return New(c.getAPIClientOptions(), optFns...)
}

// Create presign client from SDK configuration
presigner := rds.NewPresignClientFromConfig(cfg, func(o *rds.PresignOptions){
     o.Presigner = v4.NewPresigner()
})

// e.g. PresignClient given options, and optional presigner,
presigner := rds.NewPresignClient(options, func(o *rds.PresignOptions){
     o.Presigner = v4.NewPresigner()
})
```